### ISSUE TITLE
Problem with c++ dependencies

### ISSUE THREAD

**Issue by vladar** [DoctorEvidence/lmdb-store#46]
I am seeing the following error when trying to access `lmdb-store` package [in CircleCI](https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/63147/workflows/5f99f3fb-ded6-4faa-8e9a-cbfe42b53589/jobs/692150):

```
Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/circleci/project/node_modules/lmdb-store/build/Release/lmdb-store.node)
  
  - loader.js:1122 Object.Module._extensions..node
    internal/modules/cjs/loader.js:1122:18
  
  - loader.js:928 Module.load
    internal/modules/cjs/loader.js:928:32
  
  - loader.js:769 Function.Module._load
    internal/modules/cjs/loader.js:769:14
  
  - loader.js:952 Module.require
    internal/modules/cjs/loader.js:952:19
  
  - v8-compile-cache.js:159 require
    [project]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - index.js:21 load
    [project]/[lmdb-store]/[node-gyp-build]/index.js:21:10
  
  - index.js:7 Object.<anonymous>
    [project]/[lmdb-store]/index.js:7:49
  
  - v8-compile-cache.js:192 Module._compile
    [project]/[v8-compile-cache]/v8-compile-cache.js:192:30
  
  - loader.js:1092 Object.Module._extensions..js
    internal/modules/cjs/loader.js:1092:10
  
  - loader.js:928 Module.load
    internal/modules/cjs/loader.js:928:32
  
  - loader.js:769 Function.Module._load
    internal/modules/cjs/loader.js:769:14
  
  - loader.js:952 Module.require
    internal/modules/cjs/loader.js:952:19
  
  - v8-compile-cache.js:159 require
    [project]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - lmdb-datastore.ts:1 Object.<anonymous>
    /home/circleci/project/packages/gatsby/src/datastore/lmdb/lmdb-datastore.ts:    1:1
  
  - v8-compile-cache.js:192 Module._compile
    [project]/[v8-compile-cache]/v8-compile-cache.js:192:30
  
  - loader.js:1092 Object.Module._extensions..js
    internal/modules/cjs/loader.js:1092:10
```

It errors on this line:

```
import { ArrayLikeIterable, open, RootDatabase } from "lmdb-store"
```

Any additional installation instructions for cases like this?

**Comment by kriszyp** [comment]
This SE response suggests that updated standard C++ libs need to be updated:
https://askubuntu.com/questions/575505/glibcxx-3-4-20-not-found-how-to-fix-this-error
I am not sure if you have access to the server to be able to do that? What version of Node are you trying to run this on? Node v16 use V8 9.0, which has upgraded to requiring C++ v14. However, if the issue is that older versions of Node might be running on machines with older C++ versions, I could potentially make the required C++ version be conditional on the Node version it is targeting.

**Comment by vladar** [comment]
It is Node 14.16.0.

But it seems to be VM image-specific. I have other CircleCI jobs that run fine with the same version of Node. The only difference is that they run in Docker with different VM images while this job runs in a default Linux VM of CircleCI (so different VM image).

So, seems like I need to install the correct C++ dependencies. Here they suggest installing `gcc-4.9` and `g++-4.9`: https://stackoverflow.com/questions/20357033/usr-lib-x86-64-linux-gnu-libstdc-so-6-version-cxxabi-1-3-8-not-found

But I wonder if there is a way to check it to output some useful message to users who will see this error?

**Comment by kriszyp** [comment]
Do you think https://github.com/DoctorEvidence/lmdb-store/commit/cb151ba16a2bb679d26c3aebeef3ef3aac4684df will help? (I can publish if you think it will).

**Comment by vladar** [comment]
I have no idea 😅 Maybe you can [publish a canary](https://gist.github.com/zephraph/d97ce2b42596eb2d57227beeab6315e2) version? Then I will test it with our CircleCI and see if it helps.

**Comment by kriszyp** [comment]
I went ahead and published in v1.4.2, as there was some other code I wanted to get published as well.

**Comment by vladar** [comment]
Cool! I don't see this error anymore with `1.4.2`. So this seems to be fixed. Thank you!
